jellyfish/server_notifications.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+
+package Jellyfish.Server;
+
+message ControlMessage {
+  oneof content {
+    ServerNotification.RoomNotification roomCrashed = 1;
+    ServerNotification.PeerNotification peerConnected = 2;
+    ServerNotification.PeerNotification peerDisconnected = 3;
+    ServerNotification.PeerNotification peerCrashed = 4;
+    ServerNotification.ComponentNotification componentCrashed = 5;
+    ServerNotification.Authenticated authenticated = 6;
+
+    ClientMessage.TokenMessage authRequest=7;
+  }
+}
+
+message ClientMessage {
+  message TokenMessage{
+    string token=1;
+  }
+}
+
+message ServerNotification {
+    message Authenticated{}
+
+    message RoomNotification{
+        string roomId=1;
+    }
+
+    message PeerNotification{
+        string roomId=1;
+        string peerId=2;
+    }
+
+    message ComponentNotification{
+        string roomId=1;
+        string componentId=2;
+    }
+}
```

Comment:
I don't know if using `ServerNotification` and `ClientMessage` only to group other `message` definitions is a good idea -- now in theory empty `ServerNotification` is a valid message. I get that you can validate if some message is actually `ServerNotification` or `ClientMessage` in Elixir code, but wouldn't it be more idiomatic to do
```protobuf
message ControlMessage {
  message ServerNotification {
    message PeerCrashed {
      string peer_id = 1;
      [...]
    }
    [...]
    oneof content {
      PeerCrashed peer_crashed = 1;
    }
  }

  message ClientMessage {
    [...]
  }
  oneof content {
    ServerNotification server_notification = 1;
    ClientMessage client_message = 2;
  }
}
```
Or something like that, doesn't have to be nested ofc. I'm not sure tho.